fluid/face_detection/image_util.py

```diff
@@ -177,12 +247,41 @@ def generate_batch_samples(batch_sampler, bbox_labels, image_width,
     return sampled_bbox
 
 
-def clip_bbox(src_bbox):
-    src_bbox.xmin = max(min(src_bbox.xmin, 1.0), 0.0)
-    src_bbox.ymin = max(min(src_bbox.ymin, 1.0), 0.0)
-    src_bbox.xmax = max(min(src_bbox.xmax, 1.0), 0.0)
-    src_bbox.ymax = max(min(src_bbox.ymax, 1.0), 0.0)
-    return src_bbox
+def generate_batch_random_samples(batch_sampler, bbox_labels, image_width,
+                                  image_height, scale_array, resize_width,
+                                  resize_height):
+    sampled_bbox = []
+    index = []
+    c = 0
+    for sampler in batch_sampler:
+        found = 0
+        for i in range(sampler.max_trial):
+            if found >= sampler.max_sample:
+                break
+            sample_bbox = data_anchor_sampling(
+                sampler, bbox_labels, image_width, image_height, scale_array,
+                resize_width, resize_height)
+            if satisfy_sample_constraint(sampler, sample_bbox, bbox_labels):
+                sampled_bbox.append(sample_bbox)
+                found = found + 1
+                index.append(c)
+        c = c + 1
+    return sampled_bbox
+
+
+def clip_bbox(src_bbox, out_of_range=False):
+    if not out_of_range:
+        src_bbox.xmin = max(min(src_bbox.xmin, 1.0), 0.0)
+        src_bbox.ymin = max(min(src_bbox.ymin, 1.0), 0.0)
+        src_bbox.xmax = max(min(src_bbox.xmax, 1.0), 0.0)
+        src_bbox.ymax = max(min(src_bbox.ymax, 1.0), 0.0)
+        return src_bbox
+    else:
```

Comment:
Why this branch is needed, since it keep the same

Reply:
remote this branch.